gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubSyncDataWriter.java

```diff
@@ -0,0 +1,205 @@
+   */
+  public long bytesWritten() {
+    return bytesWritten.getCount();
+  }
+
+  /**
+   * A signature which contains the duration.
+   * After the duration is expired, the signature becomes invalid
+   */
+  private String getSignature() {
+    // generate signature
+    try {
+      String signature = SharedAccessSignatureTokenProvider
+          .generateSharedAccessSignature(sasKeyName, sasKey, namespaceName, Duration.ofMinutes(45));
+      return signature;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Send an encoded string to the Eventhub using post method
+   */
+  private CloseableHttpResponse post (String encoded) throws  IOException {
+
+    String uri = "https://" + namespaceName + ".servicebus.windows.net/" + eventHubName + "/messages";
+    HttpPost httpPost = new HttpPost(uri);
+    httpPost.setHeader("Content-type", "application/vnd.microsoft.servicebus.json");
+    httpPost.setHeader("Authorization", getSignature());
+    httpPost.setHeader("Host", namespaceName + ".servicebus.windows.net ");
+
+
+    StringEntity entity = new StringEntity(encoded);
+    httpPost.setEntity(entity);
+
+    CloseableHttpResponse response = httpclient.execute(httpPost);
+
+    try {
+      System.out.println(response.getStatusLine());
+      HttpEntity entity2 = response.getEntity();
+      // do something useful with the response body
+      // and ensure it is fully consumed
+      EntityUtils.consume(entity2);
+    } finally {
+      response.close();
+    }
+
+    return response;
+  }
+
+  /**
+   * Each record of batch is wrapped by a 'Body' json object
+   * put this new object into an array, encode the whole array
+   */
+  private String encodeBatch (Batch<JsonObject> batch) throws IOException{
+    // Convert original json object to a new json object with format {"Body": "originalJson"}
+    // Add new json object to an array and send the whole array to eventhub using REST api
+    // Refer to https://docs.microsoft.com/en-us/rest/api/eventhub/send-batch-events
+    ObjectMapper mapper = new ObjectMapper();
+    ArrayList<TestJson.Message> arrayList = new ArrayList<>();
+
+    List<Batch.RecordWithCallback<JsonObject>> list = batch.getRecords();
+    for (Batch.RecordWithCallback<JsonObject> recordAndCallback: list) {
+      JsonObject record = recordAndCallback.getRecord();
+      arrayList.add(new TestJson.Message(record.toString()));
+    }
+    return mapper.writeValueAsString (arrayList);
+  }
+
+  /**
+   * A single record is wrapped by a 'Body' json object
+   * encode this json object
+   */
+  private String encodeRecord (JsonObject record)throws  IOException {
+    // Convert original json object to a new json object with format {"Body": "originalJson"}
+    // Add new json object to an array and send the whole array to eventhub using REST api
+    // Refer to https://docs.microsoft.com/en-us/rest/api/eventhub/send-batch-events
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
Use Gson? 

Also the mapper / gson object should be a static final and reused.